app/src/main/java/nu/yona/app/ui/message/MessageStickyRecyclerAdapter.java

```diff
@@ -64,63 +71,137 @@ public MessageItemViewHolder onCreateViewHolder(ViewGroup parent, int viewType)
 	@Override
 	public void onBindViewHolder(MessageItemViewHolder holder, int position)
 	{
-		YonaMessage yonaObject = (YonaMessage) getItem(position);
+		currentYonaMessage = (YonaMessage) getItem(position);
+		currentMessageItemHolder = holder;
+		if (currentYonaMessage != null)
+		{
+			setUpMessageListItemTitle();
+			setUpMessageItemHolderDetails();
+			currentMessageItemHolder.deleteMsg.setTag(currentYonaMessage);
+			currentMessageItemHolder.messageContainer.setTag(currentYonaMessage);
+		}
+	}
 
-		if (yonaObject != null)
+	private void setUpMessageListItemTitle()
+	{
+		if (currentYonaMessage.getNotificationMessageEnum() != null && !TextUtils.isEmpty(currentYonaMessage.getNotificationMessageEnum().getUserMessage()))
 		{
-			if (yonaObject.getNotificationMessageEnum() != null && !TextUtils.isEmpty(yonaObject.getNotificationMessageEnum().getUserMessage()))
-			{
-				holder.txtTitleMsg.setText(yonaObject.getMessage());
-				holder.img_status.setImageResource(yonaObject.getNotificationMessageEnum().getImageId());
-			}
-			if (yonaObject.getLinks() != null && yonaObject.getLinks().getEdit() != null)
+			if (currentYonaMessage.getNotificationMessageEnum().getUserMessage().equals(NotificationEnum.BUDDYCONNECTREQUESTMESSAGE))
```

Comment:
This will never be true. ``currentYonaMessage.getNotificationMessageEnum().getUserMessage()`` returns a string, which will never be equal to an enum value. It should be this:
```
if (currentYonaMessage.getNotificationMessageEnum() == NotificationEnum.BUDDYCONNECTREQUESTMESSAGE)
```
(enums can be compared with ``==``).

Reply:
Code updated as the above code to set message title is not needed. 
User Message from NotificationMessageEnum alone is sufficient.